tests/tlstest.py

```diff
@@ -1638,6 +1653,22 @@ def server_bind(self):
 
     test_no += 1
 
+    print("Test {0} - Heartbeat extension response callback".format(test_no))
+    synchro.send(b'R')
```

Comment:
server should send the synchro when it is ready, as in, just before calling connect() and handshakeServer()

this means the os.urandom() definitely should be _before_ synchro.send(), setting up the callbacks probably should be also before it

in other test cases we can be a bit more forgiving as there only few values are set, which takes µs, not, potentially, seconds the os.urandom() can take